### ISSUE TITLE
symlink failed at line 102

### ISSUE THREAD

**Issue by clharrison** [kal247/App-remotediff#2]
Am running MacOS 14.5.1 and perl v5.38.2

> symlink failed : File exists at /Users/chap/perl5/perlbrew/perls/perl-5.38.0/bin/remotediff line 102

```
DB<1> c 102
main::(/Users/chap/perl5/perlbrew/perls/perl-5.38.0/bin/remotediff:102):
102:		symlink(abs_path($ARGV[0]),"$tmp/$path/$base") or die "symlink failed : $!";
  DB<5> p "$tmp/$path/$base"
/var/folders/t7/vp6s59yn5l1bxv8r07k7c_000000gs/T/remotediff/./httpd.conf
  DB<7> !ls -l /var/folders/t7/vp6s59yn5l1bxv8r07k7c_000000gs/T/remotediff/./httpd.conf
lrwxrwxr-x  1 chap  staff  31 May 26 16:47 /var/folders/t7/vp6s59yn5l1bxv8r07k7c_000000gs/T/remotediff/./httpd.conf -> /private/etc/apache2/httpd.conf

```
Deleting the symlink fixes the problem.

**Comment by kal247** [comment]
I've just fixed it in v1.14, could you please test it?

Thank you!

**Comment by clharrison** [comment]
Yes, that fixed the problem. But then I tried:

`/etc/apache2$ remotediff -r . retsina.local:/etc/apache2`

and again got 

`symlink failed : File exists at /Users/chap/perl5/perlbrew/perls/perl-5.38.0/bin/remotediff line 103`

**Comment by kal247** [comment]
`remotediff` has been thought as a wrapper over `diff` to have a 100% identical behaviour.

I wanted that the real `diff` prints the same paths as `remotediff` arguments (and not be replaced by very long and ugly paths containing the temporary directory).

There's a gap in my trivial solution, when one argument is "." (single dot).

I see two possible fixes : 

1. Forbid the use of "." like ".." are forbidden. My preferred fix.
2. As a special case, replace "." with the absolute path to CWD, before calling `diff`.

What's your opinion?

**Comment by clharrison** [comment]
I have a mild preference for making "." be acceptable, since I always navigate to a directory before diffing it with the remote version. But if you choose option 1, I'll just write a `remotediff` wrapper to replace '.' with the absolute path.

**Comment by kal247** [comment]
Since you might be the only person using this, let's go for option 2!
Fixed in v1.15, please let me know if it doesn't work.
Thanks

**Comment by clharrison** [comment]
Looks good. Thanks!
